climpred/classes.py

```diff
@@ -40,34 +40,38 @@ def _display_metadata(self):
     """
     header = f'<climpred.{type(self).__name__}>'
     summary = header + '\nInitialized Ensemble:\n'
-    summary += '    ' + str(self.initialized.data_vars)[18:].strip() + '\n'
+    summary += '    ' + str(self._datasets['initialized'].data_vars)[18:].strip() + '\n'
     if isinstance(self, HindcastEnsemble):
-        if any(self.reference):
-            for key in self.reference:
+        if any(self._datasets['reference']):
```

Comment:
Can you elaborate what this line does? I think if self._datasets['reference'] is an empty list and you're looping through, it automatically stops the iteration.

Reply:
Added a comment here. If it's an empty dict it just prints "None" for the `__repr__`. Otherwise it prints out all of the variables with some metadata for dimensions.